src/foam/nanos/actioncommand/ActionCommandDAO.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.actioncommand;
+
+import foam.core.ClassInfo;
+import foam.core.FObject;
+import foam.core.X;
+import foam.nanos.actioncommand.ActionCommand;
+import java.lang.IllegalAccessException;
+import java.lang.NoSuchMethodException;
+import java.lang.SecurityException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ActionCommandDAO
+  extends foam.dao.ProxyDAO
+{
+  public Object cmd(Object obj) {
+    if ( obj instanceof ActionCommand ) {
+      FObject actionObj = ((ActionCommand)obj).getObject();
+      try {
+        // Calling method
+        Method action = actionObj.getClass().getDeclaredMethod(((ActionCommand)obj).getActionName());
+        action.invoke(actionObj);
+      } catch ( SecurityException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+
+      }
+    }
+    return null;
+  }
+
+  public Object cmd_(X x, Object obj) {
+    this.cmd(obj);
+    return null;
+  }
+}
```

Comment:
@adamvy The idea is that the action would be applied to an object already stored in a DAO. So we could have action buttons in the GUI but when you press them the action is actually performed on the server. So in a GUI in a UpdateView you would have actions that you could trigger which would then be run on the server. The server would find the object and then call it's server-side Java action (which is really just a method there).